cmd/tikv-ctl/src/main.rs

```diff
@@ -665,9 +689,43 @@ fn compact_whole_cluster(
         handles.push(h);
     }
 
-    for h in handles {
-        h.join().unwrap();
-    }
+    handles.into_iter().for_each(|h| h.join().unwrap());
+}
+
+fn flashback_whole_cluster(
+    pd_client: &RpcClient,
+    cfg: &TikvConfig,
+    mgr: Arc<SecurityManager>,
+    region_ids: Vec<u64>,
+    version: u64,
+    start_key: Vec<u8>,
+    end_key: Vec<u8>,
+) {
+    let stores = pd_client
+            .get_all_stores(true) // Exclude tombstone stores.
+            .unwrap_or_else(|e| perror_and_exit("Get all cluster stores from PD failed", e));
+
+    let mut handles = Vec::new();
+    stores.into_iter().for_each(|s| {
+        let cfg = cfg.clone();
+        let mgr = Arc::clone(&mgr);
+        let addr = s.address;
+        let region_ids = region_ids.clone();
+        let start_key = start_key.clone();
+        let end_key = end_key.clone();
+
+        let h = thread::Builder::new()
+            .name(format!("flashback-{}", addr))
```

Comment:
What about using the store ID? The address might be too long to fit the thread name length limit.